File: server/src/main/scala/producer/message_generator/json_generator/JsJsonGenerator.scala
package producer.message_generator.json_generator

import com.tools.teal.pulsar.ui.producer.v1.producer as pb

case class JsJsonGenerator(
    jsCode: String
)

object JsJsonGenerator:
    def fromPb(v: pb.JsJsonGenerator): JsJsonGenerator =
        JsJsonGenerator(
            jsCode = v.jsCode
        )

    def toPb(v: JsJsonGenerator): pb.JsJsonGenerator =
        pb.JsJsonGenerator(
            jsCode = v.jsCode
        )

File: ui/components/ui/ProducerSession/message-generator/data-generators/json-generator/js-json-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type JsJsonGenerator = {
  type: 'js-json-generator',
  jsCode: string
};

export function jsJsonGeneratorFromPb(v: pb.JsJsonGenerator): JsJsonGenerator {
  return {
    type: 'js-json-generator',
    jsCode: v.getJsCode()
  };
}

export function jsJsonGeneratorToPb(v: JsJsonGenerator): pb.JsJsonGenerator {
  const message = new pb.JsJsonGenerator();
  message.setJsCode(v.jsCode);

  return message;
}
